src/cairoUtilFuncGen/storage/storageRead.ts

```diff
@@ -61,13 +67,30 @@ export class StorageReadGen extends StringIndexedFuncGen {
   private getOrCreate(typeToRead: CairoType): GeneratedFunctionInfo {
     const funcName = `WS${this.generatedFunctionsDef.size}_READ_${typeToRead.typeName}`;
     const resultCairoType = typeToRead.toString();
-    const [reads, pack] = serialiseReads(typeToRead, readFelt, readId);
+    let funcBody: string;
+    if (typeToRead instanceof CairoBool) {
+      const equivalentStoredType = new CairoFelt();
+      // Given that only 1 variable was generated, then, in `pack` is stored its name
+      const [reads, pack] = serialiseReads(equivalentStoredType, readFelt, readId);
+      assert(reads.length === 1, 'Storage read for Cairo Bools should generate only 1 variable');
+      // It's assumed here that `1` -> True and `0` -> False. With simplification purposes it´s
+      // infered that if is not `1` then it must be false.
+      funcBody = endent`
+        ${reads.map((s) => `  ${s}`).join('\n')}
+        ${pack} == 1
+      `;
```

Comment:
This Cairo Type is computed from the Solidity Type already in the caller function, so it's actually obtained from it. I think your requests resume to point 1 and points 2-3 are embedded in that change. I'm not sure though how it will improve extensibility. This pattern of receiving the Cairo Type in the `getOrCreate` function is not unique to `storageWrite` or `storageRead` Eg:
https://github.com/NethermindEth/warp/blob/abb2b5508acf3e19a51344d7fef6d6d7a3be4aa3/src/cairoUtilFuncGen/memory/memoryMemberAccess.ts#L82
I look at other funcGen scripts and they do both approach. If that's what you want to solve (have a unified structure for all `getOrCreate` functions) I think it could be a separate Issue

Reply:
>This Cairo Type is computed from the Solidity Type already in the caller function, so it's...

When we are dealing in the memory space, we are dealing with felts: uXs, booleans, addresses etc.. are stored as felts. When you read from those values we are no longer expecting a `felt` but a uXs, boolean or address. So the function that both read and write to the memory space should serve as translation layers to handle this logic.

You wrote a logic for bool but it is not extensible because you create a hardcoded cairo felt to generate the writings of such bool. (You could directly generate the writings for the `bool`)

Also, I am now noticing that both branches would have the same result no matter the difference 🤔 . I think there is no need to do `if is a bool` condtion. The read and write functions will work flawlessly without defining any special case for any cairo type

Reply:
The more I think of it the more I'm convinced that we should use `into()` trait for those conversions (or make a similiar one). Perhaps there is into trait that translates bool -> felt and back?